src/main/java/org/glite/authz/pep/pip/provider/policynamespip/PolicyNamesCache.java

```
@@ -0,0 +1,625 @@
+
+	// Create new Entry for this path, already setting the name and the like
+	Entry entry=new Entry(path, modified);
+
+	// Parse out the value of the key into the entry, this can throw
+	// IOException or ParseException
+	parseSubjectDNvalue(entry, value);
+
+	// Now return the entry
+	return entry;
+    }
+
+    /**
+     * Parses an {@value #FILE_SFX} file for the {@link #SUBJECT_KEY} key and
+     * returns the value.
+     * @param path Path of this info file
+     * @return String with the value of the {@link #SUBJECT_KEY}
+     * @throws IOException when reading the file failed
+     */
+    private String getSubjectDNvalue(Path path) throws IOException	{
+	String name=path.toString();
+	BufferedReader reader=null;
+	StringBuilder linebuilder=new StringBuilder();
+	String newline;
+	String value=null;
+
+	try {
+	    reader=Files.newBufferedReader(path, Charset.defaultCharset());
+	} catch (IOException e)	{
+	    throw new IOException("Cannot open "+name+": "+e.getMessage());
+	}
+	
+	// initialize line
+	try {
+	    while ( (newline=reader.readLine()) != null )   {
+		// Append to existing or empty line
+		linebuilder.append(newline);
+
+		// Handle continuation char
+		int end=linebuilder.length()-1;
+		if (end>=0 && linebuilder.charAt(end)=='\\')	{
+		    linebuilder.deleteCharAt(end);
+		    continue;
+		}
+
+		// Remove leading whitespace (easiest when converting to String)
+		String line=linebuilder.toString().trim();
+		// Only look at non-empty non-comment lines
+		if (!line.isEmpty() && line.charAt(0)!='#')	{
+		    // Split into key / value and look for subjectdn
+		    int sep=line.indexOf('=');
+		    if (sep>=0 && SUBJECT_KEY.equals(line.substring(0,sep).trim())) {
+			value=line.substring(sep+1).trim();
+			break;
+		    }
+		}
+
+		// Continue with next line
+		linebuilder.setLength(0);
+	    }
+	} catch (IOException e)	{
+	    // Try to close, this might throw a new IOException. We're throwing
+	    // one in any case.
+	    reader.close();
```

the potential exception here should be swallowed, in order to properly report the error in the next line

---

agreed, done